src/sentry/db/mixin.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+
+import six
+
+
+from uuid import uuid4
+
+from sentry.models import OrganizationOption
+
+
+def delete_pending_deletion_option(instance, **kwargs):
+    if not issubclass(instance.__class__, PendingDeletionMixin):
+        return
+    instance.delete_pending_deletion_option()
+
+
+class PendingDeletionMixin(object):
+    _rename_fields_on_pending_delete = []
+
+    def build_pending_deletion_key(self):
+        return 'pending-delete:%s:%s' % (self.__class__, self.id)
```

Comment:
Is there any possibility of hitting the limit here? https://github.com/getsentry/sentry/blob/master/src/sentry/models/organizationoption.py#L122

Reply:
Perhaps on uuid columns, but integers should be safe.

Reply:
Do you want `self.__class__.__name__` Right now you'll get the `__str__` of the class which will contain some python noise.